docs/Decorators.md

```diff
@@ -56,6 +56,27 @@ fastify.decorateRequest('utility', function () {
 
 Note: using an arrow function will break the binding of `this` to the Fastify `request` instance.
 
+<a name="usage"></a>
+#### Usage Notes
+`decorateReply` and `decorateRequest` are used to modify the `Reply` and `Request` constructors respectively by adding methods or properties. To update these properties you should directly access the desired property of the `Reply` or `Request` object.
+
+As an example let's add a user property to the `Request` object:
+
+```js
+// Decorate request with a 'user' property
+fastify.decorateRequest('user', '') 
+
+// Update our property
+fastify.addHook('preHandler', (req, reply) => {
```

Comment:
You should declare this hook as async or add `next` :)

Reply:
Ah yes. Added.